www/js/metrics/CarbonTextCard.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useMemo } from 'react';
+import { View } from 'react-native';
+import { Card, Text, useTheme} from 'react-native-paper';
+import { MetricsData } from './metricsTypes';
+import { cardStyles } from './MetricsTab';
+import { useTranslation } from 'react-i18next';
+import { formatDateRangeOfDays, parseDataFromMetrics, generateSummaryFromData, calculatePercentChange, segmentDaysByWeeks } from './metricsHelper';
+import { getAngularService } from '../angular-react-helper';
+
+type Props = { userMetrics: MetricsData, aggMetrics: MetricsData }
+const CarbonTextCard = ({ userMetrics, aggMetrics }: Props) => {
+
+  const { colors } = useTheme();
+  const { t } = useTranslation();
+  const FootprintHelper = getAngularService("FootprintHelper");
+
+  const userText = useMemo(() => {
+    if(userMetrics?.distance?.length > 0) {
+        //separate data into weeks
+        const [thisWeekDistance, lastWeekDistance] = segmentDaysByWeeks(userMetrics?.distance, 2);
```

Comment:
instead of recomputing this separately for `CarbonTextCard` and `CarbonFootprintCard`, is it possible to compute it once in `metricsHelper`? This may not be too much of a performance optimization but I want to get the sense of whether it is possible in react and is "the react way".

Reply:
Certainly possible. The body of a `useMemo` can call any function, including ones exported from generic Typescript modules ('helpers' as I've been naming them)
TODO cleanup